src/main/java/net/dries007/tfc/common/recipes/InstantBarrelRecipe.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed under the EUPL, Version 1.2.
+ * You may obtain a copy of the Licence at:
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ */
+
+package net.dries007.tfc.common.recipes;
+
+import com.google.gson.JsonObject;
+import net.minecraft.network.FriendlyByteBuf;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.item.crafting.RecipeSerializer;
+import net.minecraft.world.item.crafting.RecipeType;
+import net.minecraft.world.level.Level;
+
+import net.dries007.tfc.common.blockentities.BarrelBlockEntity;
+import org.jetbrains.annotations.Nullable;
+
+public class InstantBarrelRecipe extends BarrelRecipe
+{
+    public InstantBarrelRecipe(ResourceLocation id, Builder builder)
+    {
+        super(id, builder);
+    }
+
+    @Override
+    public boolean matches(BarrelBlockEntity.BarrelInventory container, @Nullable Level level)
+    {
+        // Instant recipes must have enough input items to convert fully, all the fluid. Excess items will get placed into the overflow
```

Comment:
Modify this comment to properly reflect the changed behavior
```suggestion
        // Instant recipes change behavior depending on the fluid content. If the recipe has no input fluid, or has no output fluid, it behaves as normal. Otherwise, it must have enough input items to fully consume all input fluid.
```